packages/h5wasm/src/utils.ts

```diff
@@ -241,6 +245,7 @@ export function parseDType(metadata: Metadata): DType {
   if (h5tClass === H5TClass.Array) {
     const { array_type } = metadata;
     assertDefined(array_type);
+    assertNonNull(array_type.shape);
```

Comment:
Expectedly, we now need to assert that the shape is non-null in a couple of places.